node/resolvers/sponsoredProducts.ts

```diff
@@ -63,11 +64,17 @@ export async function sponsoredProducts(
 ): Promise<SponsoredProduct[]> {
   if (!showSponsoredProducts(args)) return []
 
-  const adResponse = await ctx.clients.adServer.getSponsoredProducts({
-    count: SPONSORED_PRODUCTS_COUNT,
-    searchParams: getSearchParams(args),
-    userId: args.anonymousId,
-  })
+  try {
+    const adResponse = await ctx.clients.adServer.getSponsoredProducts({
+      count: SPONSORED_PRODUCTS_COUNT,
+      searchParams: getSearchParams(args),
+      userId: args.anonymousId,
+    })
 
-  return mapSponsoredProduct(adResponse)
+    return mapSponsoredProduct(adResponse)
+  } catch (error) {
+    if (error.response?.data === AdServer.ERROR_MESSAGES.AD_NOT_FOUND) return []
```

Comment:
@rodorgas I don't like this, since there may be more reasons for a 404 - for instance, a route that doesn't exist. In this case, the resolver should return a 500, since it's not an expected error.

At most, I could also check the status code as well as the message, but I feel this is not necessary.

Reply:
You're right. Maybe Ad Server should not be returning 404 in that case, could be an empty list or 204 with empty body. Either way, not an issue on this PR.